packages/ketcher-core/src/application/ketcher.ts

```diff
@@ -714,6 +719,13 @@ export class Ketcher {
     );
 
     editor.updateMonomersLibrary(dataInKetFormat);
+    if (SettingsManager.persistMonomerLibraryUpdates) {
+      const updateString =
+        typeof dataInKetFormat === 'string'
+          ? dataInKetFormat
+          : JSON.stringify(dataInKetFormat);
```

Comment:
This check is used in a few places in code right now, specifically it's used in `ensureMonomersLibraryDataInKetFormat` function
Maybe we could use or create some utility function for this?